pkg/build/builder/cmd/builder.go

```diff
@@ -32,11 +33,22 @@ func run(builderFactory factoryFunc) {
 	}
 	buildStr := os.Getenv("BUILD")
 	build := api.Build{}
-	err = json.Unmarshal([]byte(buildStr), &build)
-	if err != nil {
+	if err := latest.Codec.DecodeInto([]byte(buildStr), &build); err != nil {
 		log.Fatalf("Unable to parse build: %v", err)
 	}
-	authcfg, authPresent := dockercfg.NewHelper().GetDockerAuth(build.Parameters.Output.Registry)
+
+	var (
+		authcfg     docker.AuthConfiguration
+		authPresent bool
+	)
+	if len(build.Parameters.Output.DockerImageReference) != 0 {
```

Comment:
this question might be answered later as i read the code but i'll ask it now so i don't forget....  is DockerImageReference now being automatically populated from the To field for buildconfigs that define a To and not a DockerImageReference value?